core-rust/state-manager/src/state_computer.rs

```diff
@@ -1210,12 +1271,55 @@ where
         self.committed_transactions_metrics
             .update(transactions_metrics_data);
 
+        let mut locked_protocol_state = self.protocol_state.write();
+        *locked_protocol_state = new_protocol_state;
+        drop(locked_protocol_state);
```

Comment:
Rather than using `drop` better to just scope:
```
    {
        let mut locked_protocol_state = self.protocol_state.write();
        *locked_protocol_state = new_protocol_state;
    }
```

Reply:
I am still very new to Rust, so maybe I am not aware of all the consequences, but to me `drop` has advantages:
- [syntax] scope indents code
  - if you do a lot of it, your code visually moves too much to the right [just inconvenience]
  - also, my eyes kinda expect `if` or `for` whenever they see indented section, and a scope confuses them
- AFAIU, `drop` gives more control (e.g. over the order in which things are dropped)
- `drop` also seems very explicit: the intention is super-clear, someone wanted to free this resource at this very point (I mean: I understand that a scope also unambiguously does it, but feels more "silent", and I often witnessed an additional code comment explaining what the scope drops - while `drop` is so obvious that is does not need commenting)